src/rust/mangaworld/template/src/template.rs

```diff
@@ -0,0 +1,243 @@
+			"Più letti" => format!("{base_url}/archive?sort=most_read&page={page}"),
+			"Più recenti" => format!("{base_url}/archive?sort=newest&page={page}"),
+			_ => format!("{base_url}/archive?page={page}"),
+		}
+	} else {
+		base_url
+	};
+	let mut count = 0;
+	let mut mangas: Vec<Manga> = Vec::new();
+	let html = Request::new(list_url, HttpMethod::Get).html()?;
+	for manga in html.select(".comics-grid .entry").array() {
+		let manga_node = manga.as_node().expect("Failed to get manga as node");
+		let title = manga_node.select(".manga-title").text().read();
+		let url = manga_node.select(".manga-title").attr("href").read();
+		let id = String::from(url.substring_after("manga/").unwrap_or_default());
+		let cover = manga_node.select("img").attr("src").read();
+		let manga_type = manga_node.select(".genre a").text().read();
+		let viewer = match manga_type.as_str() {
+			"Manga" => MangaViewer::Rtl,
+			"Manhwa" => MangaViewer::Scroll,
+			_ => MangaViewer::Ltr,
+		};
+		mangas.push(Manga {
+			id,
+			cover,
+			title,
+			author: String::new(),
+			artist: String::new(),
+			description: String::new(),
+			url,
+			categories: Vec::new(),
+			status: MangaStatus::Unknown,
+			nsfw: MangaContentRating::Safe,
+			viewer,
+		});
+		count += 1;
+	}
+	let has_more = count == 16;
+
+	Ok(MangaPageResult {
+		manga: mangas,
+		has_more,
+	})
+}
+
+pub fn parse_manga_details(base_url: String, id: String) -> Result<Manga> {
+	let url = format!("{base_url}/manga/{id}");
+	let html = Request::new(&url, HttpMethod::Get).html()?;
+	let title = html.select("h1").text().read();
+	let cover = html.select(".single-comic .thumb img").attr("src").read();
+	let author = html
+		.select("div.info > div.meta-data.row.px-1 > div:nth-child(3) a")
+		.array()
+		.map(|val| val.as_node().expect("Failed to get author").text().read())
+		.collect::<Vec<String>>()
+		.join(", ");
+	let artist = html
+		.select("div.info > div.meta-data.row.px-1 > div:nth-child(4) a")
+		.array()
+		.map(|val| val.as_node().expect("Failed to get author").text().read())
+		.collect::<Vec<String>>()
+		.join(", ");
+	let description = html.select("#noidungm").text().read();
+	let categories = html
+		.select("div.info > div.meta-data.row.px-1 > div:nth-child(2) a")
+		.array()
+		.map(|val| val.as_node().expect("Failed to get author").text().read())
+		.collect::<Vec<String>>();
+	let status = manga_status(
+		html.select("div.info > div.meta-data.row.px-1 > div:nth-child(6) > a")
+			.text()
+			.read(),
+	);
+	let nsfw = if categories
+		.iter()
+		.any(|v| *v == "Ecchi" || *v == "Hentai" || *v == "Maturo")
+		|| base_url == "https://www.mangaworldadult.com"
+	{
+		MangaContentRating::Nsfw
+	} else {
+		MangaContentRating::Safe
+	};
+	let manga_type = html
+		.select("div.info > div.meta-data.row.px-1 > div:nth-child(5) > a")
+		.text()
+		.read();
+	let viewer = match manga_type.as_str() {
+		"Manga" => MangaViewer::Rtl,
+		"Manhwa" => MangaViewer::Scroll,
+		_ => MangaViewer::Ltr,
```

Comment:
why are you falling back to ltr? you should use rtl for manga sources

Reply:
It has mostly Chinese comics so it is left to right

Reply:
Manhua should be marked as scroll. I looked at the site, and the leftover types are like Doujinshi and Oneshot, which should also be Rtl.